vllm/v1/worker/tpu_model_runner.py

```diff
@@ -576,6 +589,20 @@ def execute_model(
             input_ids = self.input_ids
             inputs_embeds = None
 
+        sampling_metadata = self.input_batch.sampling_metadata
+        num_reqs = self.input_batch.num_reqs
+        # Indices at which we sample (positions of last token in the sequence).
+        # logits_indices = logits_indices[:num_reqs]
+        padded_do_sample_indices = _pad_indices_do_sample(
+            logits_indices, num_reqs)
+        # NOTE (NickLucche) here we sync with TPU: if there's any shape
+        # mismatch in pre-processing, it will trigger a small recompilation
+        # of the code thus far. Forward graph remains untouched.
+        tpu_sampling_metadata = TPUSupportedSamplingMetadata.\
+            from_sampling_metadata(sampling_metadata, padded_do_sample_indices,
```

Comment:
There's recompilation after warmup here.

Reply:
can you share the logs to help me debug this? It's not showing on the tests we have, perhaps we can add more to prevent it from happening. I will double check with `enfore_eager=False` which is not tested rn.

Reply:
nvm see comment below.